### ISSUE TITLE
ADD get php version hint from composer.json (if present)

### ISSUE THREAD

**Issue by allanlaal** [phpenv/phpenv#114]
ADD get php version hint from composer.json (if present)

**Comment by debo** [comment]
Hi @allanlaal what do you mean with "get php version hint" exactly? Could you please elaborate further on the use case? Thank you.

**Comment by allanlaal** [comment]
composer projects have php versions in their composer.json files, so phpenv could use the oldest/newest/available PHP version from the range set there 

example composer.json paths for php:

	{
		"require": {
			"php": "^8.1"
		}
	}


	{
		"require": {
			"php": "^7.3|^8.0|^8.1"
		}
	}

**Comment by debo** [comment]
Yes I know that but what behaviour would you expect from phpenv? Do you expect it to automatically install the right version if not installed? To give a proactive warning? For `phpenv install` to automatically install the most appropriate?

**Comment by allanlaal** [comment]
yeah, an automatic phpenv install would be nice on the first run

phpenv should choose an available php version to use as a `phpenv local` value

**Comment by debo** [comment]
I suppose we could do it, however we would be the only ones in the `*env` space. Philosophically speaking `phpenv` should be as agnostic as possible, yes I know that it's very unlikely for a php project to not have a `composer.json` and so why not leverage it and create a better user experience, conversely I don't want to ask to much to the tool. @madumlao what are your thoughts on this one?

**Comment by madumlao** [comment]
This can work as a plugin. If successful, we can include it in the default phpenv installer and submit similar plugins to rbenv et al. Makes sense @debo ?

**Comment by debo** [comment]
That's actually a good idea, I didn't think about it.

**Comment by madumlao** [comment]
What needs to be worked out is what should be the plugin behavior?
- if you have a semver string, should it install the highest complying semver?
- if you already have a complying version, should it do nothing?
- should we have a separate command to install the "highest complying"?
- in fact, shouldn't we work on similar behavior for `.php-version` first as a first guess?
- if we have a consistent `composer.json` and `.php-version`, should we use that as above?

etc I think we need to iron out all those details before we can start work on this.

**Comment by madumlao** [comment]
Also commenting that with the other .env projects, such as rbenv, nenv (which I actively use), I don't know a similar behavior for the Gemfile / package.json, so if we were to put this in we would be the trend setter.

**Comment by debo** [comment]
I am an OSX user, so I can definitely help with that, but as I said and as you highlighted, for quite a while it's been a pain for us when it comes to compile PHP via `php-build`.
